### PR TITLE
test: Add rpcauth pair that generated by rpcauth.py

### LINKED ISSUE #12995: Need test for rpcauth.py
There is no test for the `share/rpcauth/rpcauth.py` script, there should be one, and it should be part of travis.

### PR DESCRIPTION
**Pull request by ken2812221**
This PR adds a rpcauth pair that is randomly generated. Also checks that rpcauth.py works fine.
Resolve #12995

### CODE REVIEW

**Comment by ken2812221** [comment]
Can this resolve #12995?

**Comment by laanwj** [comment]
> Can this resolve #12995?

Yes - looks good to me

**Review by maflcko — APPROVED** [review]
utACK. Some nits

**Comment by ken2812221** [comment]
@MarcoFalke Done.

**Comment by cdecker** [comment]
ACK

**Comment by jnewbery** [comment]
I've just skimmed the code, but I'm NACKish on this. I don't think we should be adding dependencies to the functional tests.

Better to just have a unit test for rpcauth.py.

**Comment by MarcoFalke** [comment]
@jnewbery Mind to elaborate on that. Unlike the unit tests, which are shipped in a single executable, the functional tests are only used by developers, who have all the project files.

Adding a python unit test to rpcauth would add yet another test target that is run by no one (execpt by travis, if we add it to the yaml). And the rpcauth unit test would need to borrow a ton from the functional tests anyway... I don't see the advantage.

**Comment by jnewbery** [comment]
See my comment in #12995. I don't think it's good practice to have functional tests calling out to random scripts in the source directory, and certainly isn't a good idea to hard code the relative path to that directory.

**Comment by MarcoFalke** [comment]
> certainly isn't a good idea to hard code the relative path to that directory

That could easily be fixed by adjusting `configure.ac` to create an appropriate link in the `test` directory.

**Comment by ken2812221** [comment]
Now I modify configure.ac, it can generate a symlink during ./configure

**Comment by laanwj** [comment]
> I don't think it's good practice to have functional tests calling out to random scripts in the share directory

It also calls out to "random executables" in the `src` directory. It's a good default assumption.

However, I do agree that the path needs to be override-able with an environment variable, like `BITCOIND` and `BITCOINCLI` are.

**Comment by jnewbery** [comment]
> It also calls out to "random executables" in the src directory.

No - it calls out specifically to the bitcoind, bitcoin-qt and bitcoin-cli executables, which is what the functional test framework should be testing in my opinion. I don't think expanding the functional test framework to also test scripts in the source directory is a good idea. We already drive far too much testing through the functional test framework.

**Comment by laanwj** [comment]
> Edit: I won't stand in the way if everyone else thinks this is a good idea, but using the functional test framework to also test python scripts just seems like the wrong approach to me.

I think the functional test framework should be able to test everything that is supposed to be used by the user.
So probably not developer/maintainer scripts such as `github-merge.py`, but yes this script is something of a grey area.

(in similar vein I've also wondered before why this script isn't installed on `make install`)

**Comment by laanwj** [comment]
Tested ACK 8b8032e283ae1dff00de71e2d9d28c0df626fc23, also verified that it works for out-of-tree builds.

**Comment by MarcoFalke** [comment]
utACK 8b8032e

File: test/functional/rpc_users.py [hunk 1]
```diff
@@ -143,6 +189,7 @@ def run_test(self):
         conn.close()
 
         #Wrong password for rpcuser
+        self.log.info('Wrong...')
```
Comment (kanzure) [inline test/functional/rpc_users.py]: this is not very descriptive ("wrong")

**Pull request merged**